### PR TITLE
Fix adjacent vertex filter stuck

### PR DESCRIPTION
**Pull request by mad**
https://github.com/JanusGraph/janusgraph/issues/1500

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there an issue associated with this PR? Is it referenced in the commit message?
- [ ] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [ ] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by mad** [comment]
Why this test is failed? 

`ConfiguredGraphFactoryTest.graphConfigurationShouldBeWhatWeExpectWhenUsingTemplateConfiguration`

No change that may affect with this

**Comment by porunov** [comment]
@mad I've restarted failed tests

**Comment by FlorianHockmann** [comment]
@mad That's a test that is known to be flaky: #1495
We unfortunately have [some of those flaky tests](https://github.com/JanusGraph/janusgraph/labels/testing%2Fflaky). We used to just execute all jobs that failed up to 3 times on Travis to avoid failing builds due to flaky tests, but now we can just [mark them as Flaky ](https://github.com/JanusGraph/janusgraph/blob/master/TESTING.md#marking-tests-as-flaky)which means that the test is executed a few times, allowing it to fail in some of these runs.
But we should of course look into why a test is flaky and then fix that issue. That is why we create issues for these flaky tests. Marking them as flaky is just a workaround.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/AdjacentVertexFilterOptimizerStrategy.java [hunk 1]
```diff
@@ -14,6 +14,7 @@
 
 package org.janusgraph.graphdb.tinkerpop.optimize;
 
+import org.apache.tinkerpop.gremlin.process.traversal.step.util.EmptyStep;
```
Comment (porunov) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/AdjacentVertexFilterOptimizerStrategy.java]: Move this import on the line 22. I.e. after:
`import org.apache.tinkerpop.gremlin.process.traversal.Step;`

I know that it look strange but current style is to use tinkerpop imports below janusgraph imports. I don't know that is the purpose of this style, I don't get it, but it is used right now.

**Comment by porunov** [comment]
@mad Ah, I forgot. Could you please also retarget this bugfix to `0.3` branch?

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/AdjacentVertexFilterOptimizerStrategy.java (hunk 1 above)
Comment (mad) [reply janusgraph-core/src/main/java/org/janusgraph/graphdb/tinkerpop/optimize/AdjacentVertexFilterOptimizerStrategy.java]: Ok, I resort imports

**Comment by porunov** [comment]
Merging this fix with a lazy consensus.

**Pull request merged**